src/components/relative.js

```diff
@@ -8,21 +8,112 @@ import React, {Component, PropTypes} from 'react';
 import {intlShape, relativeFormatPropTypes} from '../types';
 import {invariantIntlContext, shouldIntlComponentUpdate} from '../utils';
 
+const SECOND = 1000;
+const MINUTE = 1000 * 60;
+const HOUR   = 1000 * 60 * 60;
+const DAY    = 1000 * 60 * 60 * 24;
+
+// The maximum timer delay value is a 32-bit signed integer.
+// See: https://mdn.io/setTimeout
+const MAX_TIMER_DELAY = 2147483647;
+
+function selectUnits(delta) {
+    let absDelta = Math.abs(delta);
```

Comment:
I might switch to [this style](http://www.2ality.com/2015/02/es6-scoping.html#when_should_i_use_%3Ccode%3Elet%3C%2Fcode%3E_when_%3Ccode%3Econst%3C%2Fcode%3E%3F), but will do so in another PR.